systemtest/src/test/java/io/strimzi/systemtest/operators/FeatureGatesST.java

```diff
@@ -0,0 +1,112 @@
+
+        testEnvVars.add(new EnvVar(Environment.STRIMZI_FEATURE_GATES_ENV, "+ControlPlaneListener", null));
+        deployClusterOperatorWithEnvVars(extensionContext, testEnvVars);
+
+        resourceManager.createResource(extensionContext, KafkaTemplates.kafkaPersistent(clusterName, kafkaReplicas).build());
+
+        LOGGER.info("Check for presence of ContainerPort 9090/tcp (tcp-ctrlplane) in first Kafka pod.");
+        final Pod kafkaPod = PodUtils.getPodsByPrefixInNameWithDynamicWait(NAMESPACE, clusterName + "-kafka-").get(0);
+        ContainerPort expectedControlPlaneContainerPort = new ContainerPort(9090, null, null, "tcp-ctrlplane", "TCP");
+        List<ContainerPort> kafkaPodPorts = kafkaPod.getSpec().getContainers().get(0).getPorts();
+        assertTrue(kafkaPodPorts.contains(expectedControlPlaneContainerPort));
+
+        LOGGER.info("Try to send some messages to Kafka over the next {} seconds.", sendingTime);
+        // use example clients for jobs - sending messages for 3-5mins
+        KafkaBasicExampleClients kafkaBasicClientJob = new KafkaBasicExampleClients.Builder()
+                .withProducerName(producerName)
+                .withConsumerName(consumerName)
+                .withBootstrapAddress(KafkaResources.plainBootstrapAddress(clusterName))
+                .withTopicName(topicName)
+                .withMessageCount(sendingTime)
+                .withDelayMs(1000)
+                .withNamespaceName(NAMESPACE)
+                .build();
+
+        resourceManager.createResource(extensionContext, kafkaBasicClientJob.producerStrimzi().build());
+        resourceManager.createResource(extensionContext, kafkaBasicClientJob.consumerStrimzi().build());
+
+        LOGGER.info("Remove kafka-broker service and wait for it's recreation (should be done within 30secs)");
+        KubeClusterResource.kubeClient().deleteService(KafkaResources.brokersServiceName(clusterName));
+        ServiceUtils.waitForServiceDeletion(KafkaResources.brokersServiceName(clusterName), NAMESPACE);
+        ServiceUtils.waitForServiceCreation(KafkaResources.brokersServiceName(clusterName), NAMESPACE);
+
+        LOGGER.info("Force Rolling Update of Kafka by upscaling replicas to {}", kafkaReplicas + 2);
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(clusterName, k -> k.getSpec().getKafka().setReplicas(kafkaReplicas + 2), NAMESPACE);
+        StatefulSetUtils.waitForAllStatefulSetPodsReady(NAMESPACE, KafkaResources.kafkaStatefulSetName(clusterName), kafkaReplicas + 2, ResourceOperation.getTimeoutForResourceReadiness(Constants.STATEFUL_SET));
+
+        LOGGER.info("Waiting for clients to finish sending/receiving messages");
+        ClientUtils.waitForClientSuccess(consumerName, NAMESPACE, MESSAGE_COUNT);
+        ClientUtils.waitForClientSuccess(consumerName, NAMESPACE, MESSAGE_COUNT);
+        JobUtils.deleteJobWithWait(NAMESPACE, consumerName);
+        JobUtils.deleteJobWithWait(NAMESPACE, consumerName);
```

Comment:
You use consumer in both lines. Do we also need to delete them manually? I think `ResourceManager` should do it automatically.

Reply:
I fixed that locally, but hasn't pushed. I rather delete jobs to be sure.